tests/test_checksum.py

```diff
@@ -0,0 +1,21 @@
+import rasterio
+
+
+def test_checksum_band():
+    with rasterio.open('tests/data/RGB.byte.tif') as src:
+        checksums = [src.checksum(i) for i in src.indexes]
+        assert checksums == [25420, 29131, 37860]
+
+
+def test_checksum_band_window():
+    with rasterio.open('tests/data/RGB.byte.tif') as src:
+        window = ((0, src.height), (0, src.width))
```

Comment:
Right, but that's at the other extreme.  I was fishing for something in the middle, where you'd get nonzero values but different than the first test...